aws/resource_aws_ssm_patch_group_test.go

```diff
@@ -47,7 +47,7 @@ func testAccCheckAWSSSMPatchGroupExists(n string) resource.TestCheckFunc {
 		}
 
 		for _, i := range resp.Mappings {
-			if *i.BaselineIdentity.BaselineId == rs.Primary.Attributes["baseline_id"] && *i.PatchGroup == rs.Primary.ID {
+			if *i.BaselineIdentity.BaselineId == rs.Primary.Attributes["baseline_id"] && fmt.Sprintf("%s:%s", *i.PatchGroup, *i.BaselineIdentity.BaselineId) == rs.Primary.ID {
```

Comment:
Likewise here re function to generate ID. 

Perhaps also the conditional needs to be abstracted into a function for readability.

Reply:
Moved the comparison into a function (more than happy to modify the function name if you have a better suggestion), and fixed up the missed check in testAccCheckAWSSSMPatchGroupDestroy.